celery/app/task.py

```diff
@@ -145,6 +145,11 @@ def as_execution_options(self):
             'replaced_task_nesting': self.replaced_task_nesting,
             'origin': self.origin,
         }
+        if hasattr(self, 'stamps') and hasattr(self, 'stamped_headers'):
```

Comment:
after the integration tests, can some unit tests added here?

Reply:
Done - unit test pushed.
Thanks for reviewing quickly! ❤️